lib/puppet/defaults.rb

```diff
@@ -854,11 +854,23 @@ def self.default_diffargs
         This is distinct from the certificate authority's CRL."
     },
     :certificate_revocation => {
-        :default  => true,
-        :type     => :boolean,
-        :desc     => "Whether certificate revocation should be supported by downloading a
-          Certificate Revocation List (CRL)
-          to all clients.  If enabled, CA chaining will almost definitely not work.",
+        :default  => 'chain',
```

Comment:
This matches the semantics of the Apache [`SSLCARevocationCheck`](https://httpd.apache.org/docs/2.4/mod/mod_ssl.html#sslcarevocationcheck) setting that has `chain`, `leaf`, and `none` values. Since this value is no longer a boolean value I would rather be clear about what sort of validation we're doing, IE chain vs leaf validation, rather than have a weird ternary `true`|`leaf`|`false` triple.